nixos/lib/utils.nix

```diff
@@ -3,8 +3,8 @@ pkgs: with pkgs.lib;
 rec {
 
   # Check whenever fileSystem is needed for boot
-  fsNeededForBoot = fs: fs.neededForBoot
-                     || elem fs.mountPoint [ "/" "/nix" "/nix/store" "/var" "/var/log" "/var/lib" "/etc" ];
+  baseNeededForBootList = [ "/" "/nix" "/nix/store" "/var" "/var/log" "/var/lib" "/etc" ];
```

Comment:
```suggestion
  pathsNeededForBoot = [ "/" "/nix" "/nix/store" "/var" "/var/log" "/var/lib" "/etc" ];
```

Reply:
If these are to be paths needed for boot rather than particular mount points needed for boot, should the check be that a filesystem's mount point is a parent of one of the paths in the list, rather than exactly equal to one of the paths in the list?

Reply:
This is what `elem fs.mountPoint baseNeededForBootList` does as far as I know.

Reply:
OK, but that invariant should be preserved. I'll add a comment noting that.

Reply:
Done.